### ISSUE TITLE
Bug: After selection.formatText(...) range is not properly updated

### ISSUE THREAD

**Issue by nextend** [facebook/lexical#2439]
When your selection contains partial TextNode and you remove the format the range you end up is wrong as it still marks that selection that it has the removed format.

Lexical version: 0.3.3

## Steps To Reproduce

1. Open https://codesandbox.io/s/lexical-rich-text-example-5tncvy
2. Type the following text: Hello World   
3. Select "Hello" then format to italic
4. Select the following: "o World"
5. Format to italic

## The current behavior
After you completed the final step, the editor marks the selection still as italic while it's not.

Selection:
```
{
   "anchor":{
      "__type":"text",
      "__parent":"1",
      "__key":"2",
      "__text":"Hell",
      "__format":2,
      "__style":"",
      "__mode":0,
      "__detail":0
   },
   "anchorOffset":4,
   "focus":{
      "__type":"text",
      "__parent":"1",
      "__key":"5",
      "__text":"o World ",
      "__format":0,
      "__style":"",
      "__mode":0,
      "__detail":0
   },
   "focusOffset":6
}
```

## The expected behavior
It should not mark the selection as italic as we just removed that format from the selection

Selection:
```
{
   "anchor":{
      "__type":"text",
      "__parent":"1",
      "__key":"5",
      "__text":"o World ",
      "__format":0,
      "__style":"",
      "__mode":0,
      "__detail":0
   },
   "anchorOffset":0,
   "focus":{
      "__type":"text",
      "__parent":"1",
      "__key":"5",
      "__text":"o World ",
      "__format":0,
      "__style":"",
      "__mode":0,
      "__detail":0
   },
   "focusOffset":6
}
```

**Comment by zurfyx** [comment]
I can't reproduce it but what you're seeing is not necessarily incorrect (it's just odd). Notice that the key above corresponds to the previous node (instead of `o World`) and the offset is immediately after the last character.

https://user-images.githubusercontent.com/193447/174087332-cb253093-45b1-4935-8698-08e3696fd70f.mov

**Comment by nextend** [comment]
Window 10 Chrome Version 102.0.5005.63
https://www.youtube.com/watch?v=81lIGLWiUJU

If you watch closely the selection also breaks on the second toggle. 0:14 in the video

**Comment by nextend** [comment]
I tried the fixed version at https://lexical-playground-2x1ika56e-fbopensource.vercel.app/ and I think I was able to create a similar bug.

1. Type: An extensible text editor
2. Select: An extensible
3. Toggle italic
4. Select: sible text
5. Toggle italic

Result:
Italic will be removed from the selected text, but button state indicating that it is italic.

If you toggle italic two more time on the "sible text" selection, the italic state will be removed from the button too.

https://www.youtube.com/watch?v=8AiSHN883WI
